container/build_tar.py

```diff
@@ -191,18 +228,29 @@ def add_deb(self, deb):
     Raises:
       DebError: if the format of the deb archive is incorrect.
     """
+    pkg_data_found = False
+    pkg_metadata_found = False
     with archive.SimpleArFile(deb) as arfile:
       current = arfile.next()
-      while current and not current.filename.startswith('data.'):
+      while current:
+        if current.filename.startswith('data'):
```

Comment:
Do we want startswith or an exact match here?

Reply:
We want startswith since dpkg-source has -Z option where you can specify the compression as gzip, bzip etc..
Hence extension might be different.

Reply:
Maybe os.path.splitext would be better?